src/Temporalio/Worker/WorkflowWorker.cs

```diff
@@ -0,0 +1,254 @@
+                    // LongRunning here. That means they may oversubscribe and create a new thread
+                    // just for this, is that ok?
+                    var workflowTask = Task.Factory.StartNew(
+                        () => workflow.Activate(act),
+                        default,
+                        TaskCreationOptions.LongRunning | TaskCreationOptions.DenyChildAttach | TaskCreationOptions.PreferFairness,
+                        TaskScheduler.Current);
+                    var timeoutTask = Task.Delay(deadlockTimeout);
+                    if (timeoutTask == await Task.WhenAny(workflowTask, timeoutTask).ConfigureAwait(false))
+                    {
+                        // Hit deadlock timeout, add this to deadlocked set and have it remove
+                        // itself when it finally does finish (if ever)
+                        deadlockedWorkflows[act.RunId] = workflowTask;
+                        _ = workflowTask.ContinueWith(
+                            _ => { deadlockedWorkflows.TryRemove(act.RunId, out Task? _); },
+                            TaskScheduler.Current);
+                        throw new InvalidOperationException($"Workflow with ID {act.RunId} deadlocked after {deadlockTimeout}");
+                    }
+                    comp = await workflowTask.ConfigureAwait(false);
+                }
+                else
+                {
+                    comp = new() { Successful = new() };
+                }
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, "Failed handling activation on workflow with run ID {RunId}", act.RunId);
+                // Any error here is a completion failure
+                comp = new() { Failed = new() };
+                try
+                {
+                    comp.Failed.Failure_ = Worker.Client.Options.DataConverter.FailureConverter.ToFailure(
+                        e, Worker.Client.Options.DataConverter.PayloadConverter);
+                }
+                catch (Exception inner)
+                {
+                    logger.LogError(inner, "Failed converting activation exception on workflow with run ID {RunId}", act.RunId);
+                    comp.Failed.Failure_ = new() { Message = $"Failed converting exception: {inner}" };
+                }
+            }
+
+            // Always set the run ID of the completion
+            comp.RunId = act.RunId;
+
+            // Encode the completion if there is a codec
+            if (codec != null)
+            {
+                try
+                {
+                    await WorkflowCodecHelper.EncodeAsync(codec, comp).ConfigureAwait(false);
+                }
+                catch (Exception e)
+                {
+                    logger.LogError(e, "Failed encoding completion on workflow with run ID {RunId}", act.RunId);
+                    comp.Failed = new() { Failure_ = new() { Message = $"Failed encoding completion: {e}" } };
```

Comment:
Hmm... looks like you might have both Failure and Successful set in the completion here.

Reply:
`oneof`s properly handle mutual exclusivity internally